sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala

```diff
@@ -767,6 +777,35 @@ case class InSet(child: Expression, hset: Set[Any]) extends UnaryExpression with
   override protected def withNewChildInternal(newChild: Expression): InSet = copy(child = newChild)
 }
 
+object InSet {
```

Comment:
since these classes seem pretty general I'd maybe think about moving them somewhere where other parts of the code can use them as well, something like CollationUtils maybe?

Reply:
> since these classes seem pretty general 

hmm why? So far the classes are pretty specific to `InSet`. I didn't get why do you want to make them too generic.

Reply:
I wouldn't agree that they are specific only to `InSet`, at least in my mind they are just collation aware implementations of the scala's set interface; so they can (and probably will) be used by someone other than `InSet`

Reply:
> so they can (and probably will) be used by someone other than InSet

Is is called "Premature generalization". I would prefer to avoid the anti-pattern if it is possible. Let's make the cases more generic when it is really needed.